ksql-engine/src/main/java/io/confluent/ksql/physical/PhysicalPlanBuilder.java

```diff
@@ -253,11 +266,15 @@ private QueryMetadata buildPlanForStructuredOutputNode(
 
     }
 
-    if (updateMetastore) {
-      metaStore.putSource(sinkDataSource.cloneWithTimeKeyColumns());
-    }
+    sinkSetUp(outputNode, sinkDataSource);
 
-    final QueryId queryId = sinkDataSource.getPersistentQueryId();
+    final QueryId queryId;
+    if (outputNode.isDoCreateInto()) {
+      queryId = new QueryId(sinkDataSource.getPersistentQueryId().getId() + "_"
+                            + queryIdGenerator.getNextId());
+    } else {
+      queryId = new QueryId("InsertQuery_" + queryIdGenerator.getNextId());
```

Comment:
I agree this can be improved as you mentioned. I think the query ID can be done in a separate PR since it is not just for `INSERT INTO` but also for `CSAS` and `CTAS`.